api/src/main/java/com/github/retrooper/packetevents/util/VectorMutable3f.java

```diff
@@ -0,0 +1,237 @@
+ * Copyright (C) 2022 retrooper and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.github.retrooper.packetevents.util;
+
+import com.github.retrooper.packetevents.protocol.world.BlockFace;
+import java.util.Objects;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * 3D float Vector.
+ * This vector can represent coordinates, angles, or anything you want.
+ * You can use this to represent an array if you really want.
+ *
+ * @author retrooper
+ * @since 2.7.1
+ */
+public class VectorMutable3f implements VectorInterface3f {
+    /**
+     * X (coordinate/angle/whatever you wish)
+     */
+    public float x;
+    /**
+     * Y (coordinate/angle/whatever you wish)
+     */
+    public float y;
+    /**
+     * Z (coordinate/angle/whatever you wish)
+     */
+    public float z;
+
+    /**
+     * Default constructor setting all coordinates/angles/values to their default values (=0).
+     */
+    public VectorMutable3f() {
+        this.x = 0.0f;
+        this.y = 0.0f;
+        this.z = 0.0f;
+    }
+
+    /**
+     * Constructor allowing you to set the values.
+     *
+     * @param x X
+     * @param y Y
+     * @param z Z
+     */
+    public VectorMutable3f(float x, float y, float z) {
+        this.x = x;
+        this.y = y;
+        this.z = z;
+    }
+
+    /**
+     * Constructor allowing you to specify an array.
+     * X will be set to the first index of an array(if it exists, otherwise 0).
+     * Y will be set to the second index of an array(if it exists, otherwise 0).
+     * Z will be set to the third index of an array(if it exists, otherwise 0).
+     *
+     * @param array Array.
+     */
+    public VectorMutable3f(float[] array) {
+        if (array.length > 0) {
+            x = array[0];
+        } else {
+            x = 0;
+            y = 0;
+            z = 0;
+            return;
+        }
+
+        if (array.length > 1) {
+            y = array[1];
+        } else {
+            y = 0;
+            z = 0;
+            return;
+        }
+
+        if (array.length > 2) {
+            z = array[2];
+        } else {
+            z = 0;
+        }
+    }
+
+    public float getX() {
+        return x;
+    }
+
+    public float getY() {
+        return y;
+    }
+
+    public float getZ() {
+        return z;
+    }
+
+    public boolean equals(VectorMutable3f other) {
```

Comment:
Shouldnt the parameter be the interface? (Same for every other method in this class)